.github/workflows/test-lang-rust-ci.yml

```diff
@@ -59,19 +60,22 @@ jobs:
         uses: actions/checkout@v4
 
       - name: Cache Cargo
+        id: cache-cargo
         uses: actions/cache@v3
         with:
           # these represent dependencies downloaded by cargo
           # and thus do not depend on the OS, arch nor rust version.
           path: ~/.cargo
-          key: cargo-cache1-
+          key: ${{ runner.os }}-cargo-cache2-
+
       - name: Cache Rust dependencies
+        id: cache-target
         uses: actions/cache@v3
         with:
           # these represent compiled steps of both dependencies and avro
           # and thus are specific for a particular OS, arch and rust version.
-          path: ~/target
-          key: ${{ runner.os }}-target-cache1-${{ matrix.rust }}-
+          path: lang/rust/target
+          key: ${{ runner.os }}-target-cache2-
```

Comment:
By removing `${{ matrix.rust }}`, is this part meaningful?
I think built the deps cannot be shared among `stable`, `nightly`, `beta` and `1.65.0`.
So, I'm wondering this usage of cache has little effect, and it's better not to use cache.

Reply:
Also, why `${{ hasFiles(...) }}` is not used?